### PR TITLE
WT-12218 cache_eviction_target_strategy Statistics bug fix and Optimized macro definition

### PR DESCRIPTION
**Pull request by y123456yz**
cache_eviction_target_strategy Statistics bug fix and Optimized macro definition

### CODE REVIEW

**Comment by y123456yz** [comment]
WT-12218

File: src/evict/evict_lru.c [hunk 1]
```diff
@@ -1838,10 +1838,12 @@ __evict_walk_tree(WT_SESSION_IMPL *session, WT_EVICT_QUEUE *queue, u_int max_ent
...
         WT_STAT_CONN_INCR(session, cache_eviction_target_strategy_clean);
-    else if (!F_ISSET(cache, WT_CACHE_EVICT_CLEAN)) {
+    else if (!F_ISSET(cache, WT_CACHE_EVICT_UPDATES)) {
         min_pages *= 10;
         WT_STAT_CONN_INCR(session, cache_eviction_target_strategy_dirty);
-    } else
+    }
+
+    if (F_ISSET(cache, WT_CACHE_EVICT_DIRTY) && F_ISSET(cache, WT_CACHE_EVICT_UPDATES))
         WT_STAT_CONN_INCR(session, cache_eviction_target_strategy_both_clean_and_dirty);
```
Comment (Siddhartha8899) [inline src/evict/evict_lru.c:1847]: @y123456yz thanks for this PR.

Could you help me understand the reasoning behind these changes?

**Comment by Siddhartha8899** [comment]
@y123456yz let me know your thoughts on https://github.com/wiredtiger/wiredtiger/pull/10061#discussion_r1439128707

**Comment by y123456yz** [comment]
> @y123456yz let me know your thoughts on [#10061 (comment)](https://github.com/wiredtiger/wiredtiger/pull/10061#discussion_r1439128707)

I'm sorry, I didn't catch you earlier, but this is actually as following, I have changed：
    if (F_ISSET(cache, WT_CACHE_EVICT_CLEAN) && F_ISSET(cache, WT_CACHE_EVICT_DIRTY))
        WT_STAT_CONN_INCR(session, cache_eviction_target_strategy_both_clean_and_dirty);

https://github.com/y123456yz/wiredtiger/blob/develop_cache_eviction_target_strategy_bug_fix/src/evict/evict_lru.c#L1848-L1849

**Comment by Siddhartha8899** [comment]
@y123456yz thanks, but I do not see the current check as a bug. Could you please explain the reasoning behind this change.

**Comment by y123456yz** [comment]
> @y123456yz thanks, but I do not see the current check as a bug. Could you please explain the reasoning behind this change.

hi, @Siddhartha8899 
old code:
    if (!F_ISSET(cache, WT_CACHE_EVICT_DIRTY | WT_CACHE_EVICT_UPDATES))
          //only clean, this branch case is ok.
    else if (!F_ISSET(cache, WT_CACHE_EVICT_CLEAN)) {
           //There are 3 cases of this branch: [dirty, update, update+dirty]
    } else
	    //There are 2 cases of this branch: [update, update+dirty]


Our disagreement should be whether to consider WT_CACHE_EVICT_UPDATES？

**Comment by y123456yz** [comment]
> @y123456yz thanks, but I do not see the current check as a bug. Could you please explain the reasoning behind this change.

hi, @Siddhartha8899 
old code:
```
    if (!F_ISSET(cache, WT_CACHE_EVICT_DIRTY | WT_CACHE_EVICT_UPDATES))
        //only clean
    else if (!F_ISSET(cache, WT_CACHE_EVICT_CLEAN)) {
        //There are 3 cases of this branch: [dirty, update, update+dirty]
    } else
	    //There are 3 cases of this branch: [clean+update, clean+diry, clean+update+dirty]
```
Our disagreement should be whether to consider WT_CACHE_EVICT_UPDATES？

**Comment by Siddhartha8899** [comment]
Hi @y123456yz we are currently reviewing this specific code base internally and would like to hold off making any changes until we have determined how to proceed. So we would like to close this PR without merging at this time. There is currently no deadline for the current code review, but we will endeavor to update the corresponding Jira ticket when appropriate.

**Pull request closed without merging**